assets/js/components/settings/SettingsActiveModules.js

```diff
@@ -39,9 +39,23 @@ const { useDispatch, useSelect } = Data;
 export default function SettingsActiveModules( { activeModule, moduleState, setModuleState } ) {
 	const [ error, setError ] = useState( undefined );
 	const [ isSaving, setIsSaving ] = useState( false );
+	// We store `initialActiveSlugs` separately to avoid
+	// layout shifts when deactivating a module as it would otherwise
+	// cause the module to be removed upon deactivation.
+	const [ initialActiveSlugs, setInitialActiveSlugs ] = useState();
+	const modules = useSelect( ( select ) => select( CORE_MODULES ).getModules() );
+	useEffect( () => {
+		// Only set initialActiveSlugs once, as soon as modules are available.
+		if ( ! modules || initialActiveSlugs !== undefined ) {
+			return;
+		}
 
+		const activeSlugs = Object.keys( modules )
+			.filter( ( slug ) => modules[ slug ].active );
+
+		setInitialActiveSlugs( activeSlugs );
+	}, [ modules ] );
```

Comment:
Nice! I wish we wouldn't need such kind of changes - at _some_ point we can hopefully take some time to explore making all the screens a real single page application so that we don't need to have those "hacks" in order to have the UI wait work navigations between screens.

Reply:
@aaemnnosttv Not for this PR, but in general: Maybe we could introduce a utility to simplify getting the "initial loaded" value of a selector? Maybe something like `useInitialSelect` that uses a mix of `useSelect`, `useState`, and `useEffect` to achieve behavior close to this (just simpler), so that the function eventually always returns the initial loaded value (i.e. the first value that `useSelect` returns which is not `undefined`).